src/Cassandra/Connection.cs

```diff
@@ -600,87 +684,107 @@ public OperationState Send(IRequest request, Action<Exception, AbstractResponse>
             {
                 Request = request
             };
-            SendQueueProcess(state, true);
+            SendQueue(state);
             return state;
         }
 
-        /// <summary>
-        /// Try to write the item provided. Thread safe.
-        /// </summary>
-        /// <param name="state">The request and callback</param>
-        /// <param name="useInlining">Determines if the current thread can be used to start sending the request</param>
-        private void SendQueueProcess(OperationState state, bool useInlining)
+        private void SendQueue(OperationState state)
+        {
+            //Store into queue
+            //if its running => do nothing
+            //if not running => start dequeueing
+            _writeQueue.Enqueue(state);
+            RunQueue(true);
+        }
+
+        private void RunQueue(bool useInlining)
         {
-            var canWrite = Interlocked.CompareExchange(ref _canWriteNext, 0, 1);
-            if (canWrite == 1)
+            var isAlreadyRunning = Interlocked.CompareExchange(ref _isWriteQueueRuning, 1, 0) == 1;
+            if (isAlreadyRunning)
             {
-                if (useInlining)
-                {
-                    //Use the current thread to start the write operation
-                    SendQueueProcessItem(state);
-                    return;
-                }
-                //Start a new task using the TaskScheduler for writing
-                Task.Factory.StartNew(() => SendQueueProcessItem(state), CancellationToken.None, TaskCreationOptions.None, _writeScheduler);
+                //there is another thread writing to the wire
+                return;
             }
-            else
+            if (useInlining)
```

Comment:
Consider removing inlining (reusing the calling thread to start the writing operation) based on @alprema input. In this case, it would only improve latency for low concurrency scenarios but it doesn't make much sense to optimize for those scenarios.